group1/communication/node/main.js

```diff
@@ -29,27 +29,46 @@ var done = false;
 app.use(multer({
         'dest': '../../comm/static/uploads',
         onFileUploadStart: function(file) {
+            if(testNameValidation(file.originalname)){
                 console.log(file.originalname + ' is starting...');
+            } else {
+                console.log('File name is invalid');
+                return false;
+            }
         },
         onFileUploadComplete: function(file) {
                 console.log(file.fieldname + ' uploaded to ' + file.path);
                 done = true;
         }
 }));
 
+function testNameValidation(text) {
+    var format = /[ !@#$%^&*()+\-=\[\]{};':"\\|,<>\/?]/;
+    if(text.trim() == null || text.trim() == "" || format.test(text) == true) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
 app.post('/upload', function(req,res){
         if (done==true) {
                 var hash = crypto.randomBytes(20).toString('hex');
-				var directory =  path.join(homedir + '/sites/www.3blueprints.com/static/uploads/');
+                console.log(path);
+				var directory =  path.join(homedir + '/BU_MET_CS673/group1/comm/static/uploads/');
```

Comment:
Is the url being used for this variable correct?  Will this work in both local environments and on the web server?

Reply:
Actually I am not sure about this one. I was trying to find a way to reach the communication folder without hardcoding the folder name but after lots of trouble I couldn't do it. So for now I just put it so that we can move ahead with the work locally. Do you have any ideas on how to get the communication folder directly? The previous project guys had done it differently and it was not working at all.